reVX/hybrid_stats/hybrid_stats.py

```diff
@@ -0,0 +1,829 @@
+
+            out_stats = pd.concat(out_stats, axis=1)
+        else:
+            out_stats = cls._compute_stats(solar_data, wind_data, statistics,
+                                           diurnal=diurnal, month=month)
+
+        out_stats.index = sites.index.values
+
+        out_stats.index.name = 'gid'
+
+        return out_stats
+
+    @staticmethod
+    def save_stats(out_stats, out_fpath):
+        """
+        Save correlations to disk
+
+        Parameters
+        ----------
+        out_stats : pandas.DataFrame
+            Table of correlations to save
+        out_path : str
+            Directory, .csv, or .json path to save statistics too
+        """
+        if out_fpath.endswith('.csv'):
+            out_stats.to_csv(out_fpath)
+        elif out_fpath.endswith('.json'):
+            out_stats.to_json(out_fpath)
+        else:
+            msg = ("Cannot save statistics, expecting a directory, .csv, or "
+                   ".json path, but got: {}".format(out_fpath))
+            logger.error(msg)
+            raise OSError(msg)
+
+    def _pre_flight_check(self):
+        """
+        Compare solar and wind site meta data and time index to ensure
+        they can be compared
+
+        Returns
+        -------
+        meta : pandas.DataFrame
+            Meta data table mapping the finer resolution data to the coarser
+        time_index : pandas.DatatimeIndex
+            Coincident datetime index between solar and wind datasets,
+            i.e. datetime steps that are in both wind and solar data
+        solar_time_slice : slice | ndarray
+            slice or boolean index of the solar timesteps that are in the
+            coincident time_index
+        wind_time_slice : slice | ndarray
+            slice or boolean index of the wind timesteps that are in the
+            coincident time_index
+        """
+        with self.res_cls(self._solar_h5, hsds=self._hsds) as f:
+            solar_meta = f.meta
+            solar_ti = f.time_index
+
+        with self.res_cls(self._wind_h5, hsds=self._hsds) as f:
+            wind_meta = f.meta
+            wind_ti = f.time_index
+
+        time_index, solar_time_slice, wind_time_slice = \
+            self._check_time_index(solar_ti, wind_ti)
+        meta = self._map_sites(solar_meta, wind_meta)
+
+        return meta, time_index, solar_time_slice, wind_time_slice
+
+    def _check_stats(self, statistics):
+        """
+        check desired statistics to make sure inputs are valid
+
+        Parameters
+        ----------
+        statistics : str | tuple | dict
+            Statistics to extract, either a key or tuple of keys in
+            cls.STATS, or a dictionary of the form
+            {'stat_name': {'func': *, 'kwargs: {**}}}
+
+        Returns
+        -------
+        stats : dict
+            Dictionary of statistic functions/kwargs to run
+        """
+        STATS = {'pearson': {'func': self._pearson_correlation},
```

Comment:
Make this a class attribute? Would be nice to be able to easily see what stats are available.

Reply:
I modified this code from TemporalStats, where I do just this. BUT i wasn't sure how to do that since in this case the functions are staticmethods. I guess I could make them pure functions outside of the class